doc/internal-dev-info/how-pi4a-uses-pip.md

```diff
@@ -0,0 +1,157 @@
+
+# How python-for-android uses `pip`
+
+*Last update: July 2019*
+
+This document is meant to provide a quick summary how
+p4a (=python-for-android) uses pip and python packages in
+its build process.
+It is written to hopefully make sense from a python
+packagers point of view, not for regular end users.
+
+Please note this document just attempts to neutrally list the
+current mechanisms without endorsing or criticizing them, so
+some of this isn't necessarily meant to stay as it is but just
+how things work inside p4a in this very moment.
+
+
+## Basic concepts
+
+p4a builds & packages a python application for use on Android.
+It does this by providing a Java wrapper, and for graphical applications
+an SDL2-based wrapper which can be used with the kivy UI toolkit if
+desired (or alternatively just plain PySDL2).
+
+Any such python application will have dependencies, obviously.
+
+p4a supports two types of package dependencies for a project:
+
+**Recipe:** install script in custom p4a format. Can either install
+C/C++ or other things that cannot be pulled in via pip, or things
+that can be installed via pip but break on android by default.
+These are maintained inside the p4a source tree by p4a contributors
```

Comment:
Strictly they're only primarily maintained inside the p4a source tree, local recipes do work fine.

Reply:
Also missing full stop :p